reactor-core/src/main/java/reactor/core/publisher/Flux.java

```diff
@@ -8778,19 +8778,57 @@ public final void subscribe(Subscriber<? super T> actual) {
 	 *
 	 * @return a new {@link Flux}
 	 * @see #transform(Function) transform() for immmediate transformation of Flux
+	 * @see #transformDeferred(BiFunction) transformDeferred(BiFunction) for a similarly deferred transformation of Flux reading the ContextView
 	 * @see #as as() for a loose conversion to an arbitrary type
 	 */
 	public final <V> Flux<V> transformDeferred(Function<? super Flux<T>, ? extends Publisher<V>> transformer) {
 		return defer(() -> {
 			if (Hooks.DETECT_CONTEXT_LOSS) {
 				@SuppressWarnings({"rawtypes", "unchecked"})
-				CorePublisher<V> result = new ContextTrackingFunctionWrapper<T, V>((Function) transformer).apply(this);
-				return result;
+				ContextTrackingFunctionWrapper<T, V> wrapper = new ContextTrackingFunctionWrapper<T, V>((Function) transformer);
+				return wrapper.apply(this);
 			}
 			return transformer.apply(this);
 		});
 	}
 
+	/**
+	 * Defer the given transformation to this {@link Flux} in order to generate a
+	 * target {@link Flux} type. A transformation will occur for each
+	 * {@link Subscriber}. In addition, the transfoming {@link BiFunction} exposes
+	 * the {@link ContextView} of each {@link Subscriber}. For instance:
+	 *
+	 * <blockquote><pre>
+	 * Flux&lt;T> fluxLogged = flux.transformDeferred((ctx, original) -> original.log("for RequestID" + ctx.get("RequestID"))
+	 * //...later subscribe. Each subscriber has its Context with a RequestID entry
+	 * fluxLogged.subscriberContext(Context.of("RequestID", "requestA").subscribe();
+	 * fluxLogged.subscriberContext(Context.of("RequestID", "requestB").subscribe();
+	 * </pre></blockquote>
+	 * <p>
+	 * <img class="marble" src="doc-files/marbles/transformDeferredForFlux.svg" alt="">
+	 *
+	 * @param transformer the {@link BiFunction} to lazily map this {@link Flux} into a target {@link Flux}
+	 * instance upon subscription, with access to {@link ContextView}
+	 * @param <V> the item type in the returned {@link Publisher}
+	 * @return a new {@link Flux}
+	 * @see #as as() for a loose conversion to an arbitrary type
+	 * @see #transform(Function)
+	 * @see #transformDeferred(Function)
+	 */
+	public final <V> Flux<V> transformDeferred(BiFunction<? super ContextView, ? super Flux<T>, ? extends Publisher<V>> transformer) {
```

Comment:
can we please swap `ContextView` and `? super Flux<T>`, so that the publisher comes first and `Context` is a secondary concept.

Reply:
done